src/scripts/utils.py

```diff
@@ -0,0 +1,299 @@
+    """Starting from `min_id`, count up and check until finding the next ID.
+
+    :returns: (int) next mondo ID; (Set[int]) New set of Mondo IDs, assuming 'next mondo ID' is included."""
+    next_id = int(min_id)
+    while True:
+        next_id = next_id + 1
+        if next_id > max_id:
+            raise ValueError('Ran out of valid IDs to assign for new slurpable Mondo terms. Either `min_id`, `max_id`, '
+                             'or both should be changed to allow for new assignments.')
+        if next_id not in mondo_ids:
+            break
+    mondo_ids.add(next_id)
+    return next_id, mondo_ids
+
+
+def get_mondo_term_ids(mondo_terms_path: str, slurp_dir_path: str) -> Set[int]:
+    """From path to file of mondo terms, get set of Mondo IDs as integers.
+    # todo: Consider using `curies` package, though needing another prefix_map only for this seems even less optimal
+    """
+    # Get mondo IDs from Mondo
+    mondo_base_uri = 'http://purl.obolibrary.org/obo/MONDO_'
+    mondo_termlist_df = pd.read_csv(mondo_terms_path, comment='#', sep='\t')
+    mondo_term_uris: List[str] = list(mondo_termlist_df['?term'])
+    mondo_ids: List[int] = []
+    for x in mondo_term_uris:
+        x = x[1:] if x.startswith('<') else x
+        x = x[:-1] if x.endswith('>') else x
+        if not x.startswith(mondo_base_uri):
+            continue
+        mondo_ids.append(int(x.replace('http://purl.obolibrary.org/obo/MONDO_', '')))
+
+    # Get interim slurp-assigned Mondo IDs
+    paths = glob(os.path.join(slurp_dir_path, '*.tsv'))
+    for path in paths:
+        df = pd.read_csv(path, sep='\t', comment='#')
+        ids = list(df['mondo_id'][1:])
+        mondo_ids += ids
+        print()
+
+    return set(mondo_ids)
+
+
+    # TODO: Improve/replace 'direct_parent_curies' method: Take a look at: ProntoImpl hierararchical_parents()
+def _get_direct_owned_parent_uris(
+    ontology: ProntoImplementation, owned_prefix_map: Dict[PREFIX, URI], curie: CURIE
+) -> List[URI]:
+    """Get URIs of direct parents of a class. Only returns parents that are 'owned' by the ontology.
+
+    ontology: Haven't decided yet which implementation I'll use. Would use superclass, but they use mult inheritance.
+    owned_prefix_map: All the prefixes that are 'owned' by the ontology. Keys are CURIE prefixes and values are URIs.
+    """
+    # These vars are here for stability reasons, just in case I get CURIES where I expect URIs or vice versa.
+    subclass_preds = [x + 'subClassOf' for x in ['rdfs:', 'http://www.w3.org/2000/01/rdf-schema#']]
+    owned_prefixes = set(owned_prefix_map.keys())
+    uri_converter = curies.Converter.from_prefix_map(owned_prefix_map)
+
+    direct_owned_parent_uris: List[URI] = []
+    # todo: how to combine theses?
+    # todo: diff results? how ot chck?
+    rels1: List[CURIE] = ontology.hierararchical_parents(curie)
+    rels: List[TRIPLE] = [x for x in ontology.relationships(subjects=[curie])]
```

Comment:
@matentzn Tried to do `ontology.hierararchical_parents(curie)` as you suggested, but got an error. I made a PR with a fix: https://github.com/INCATools/ontology-access-kit/pull/287

Reply:
You can close this whenever you see it, as I'm now getting parents using SPARQL. Just want you to be aware.

Reply:
Marking this resolved